### PR TITLE
Fix KeyError in iproute pformat

### PR DESCRIPTION
**Pull request by zykovd**
## Fix KeyError in iproute pformat

This fixes KeyError on specific network configuration when running cloud-init on "network" stage.  The same problem was mentioned in #746 and #1041.
```
summary: Fix KeyError in iproute pformat

This fixes KeyError on specific network configuration when 
running cloud-init on "network" stage. 

failed run of stage init
------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/cmd/main.py", line 767, in status_wrapper
    ret = functor(name, args)
  File "/usr/lib/python3/dist-packages/cloudinit/cmd/main.py", line 367, in main_init
    sys.stderr.write("%s\n" % (netinfo.debug_info()))
  File "/usr/lib/python3/dist-packages/cloudinit/netinfo.py", line 650, in debug_info
    route_lines = route_pformat().splitlines()
  File "/usr/lib/python3/dist-packages/cloudinit/netinfo.py", line 630, in route_pformat
    r["gateway"],
KeyError: 'gateway'
------------------------------------------------------------
```

## Test Steps

1. Create machine with custom default IPv6 route, i.e.
```
root@host:~# ip -o -6 route list table all
...                   
default dev eth1 proto static metric 50 pref medium 
...
```
2. Run `cloud-init init`

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/netinfo.py [hunk 1]
```diff
@@ -598,16 +598,19 @@ def route_pformat():
...
+                        [
+                            route_id,
+                            r["destination"],
+                            r.get("gateway", "0.0.0.0"),
+                            r["genmask"],
+                            r["iface"],
+                            r["flags"],
+                        ]
+                    )
+                except Exception as e:
```
Comment (TheRealFalcon) [inline cloudinit/netinfo.py:612]: Let's use `KeyError` here instead of `Exception`

File: cloudinit/netinfo.py [hunk 2]
```diff
@@ -625,15 +628,18 @@ def route_pformat():
...
+                    tbl_v6.add_row(
+                        [
+                            route_id,
+                            r["destination"],
+                            r.get("gateway", "::"),
+                            r["iface"],
+                            r["flags"],
+                        ]
+                    )
+                except Exception as e:
```
Comment (TheRealFalcon) [inline cloudinit/netinfo.py:641]: Same comment here

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Thanks for the contribution, @zykovd ! I left small comments inline, but overall this change looks good.

I noticed you added yourself to the CLA signers file, but have you signed the [actual CLA](https://ubuntu.com/legal/contributors/agreement) too? I'm not seeing your information there. If you did, can you provide the date that you signed it?

File: cloudinit/netinfo.py (hunk 1 above)
Comment (zykovd) [reply cloudinit/netinfo.py]: Fixed

**Comment by zykovd** [comment]
> Thanks for the contribution, @zykovd ! I left small comments inline, but overall this change looks good.
> 
> I noticed you added yourself to the CLA signers file, but have you signed the [actual CLA](https://ubuntu.com/legal/contributors/agreement) too? I'm not seeing your information there. If you did, can you provide the date that you signed it?

I signed the CLA today, 05/11/2023

**Review by TheRealFalcon — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**